gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java

```diff
@@ -218,18 +221,40 @@ public class GoogleHadoopFileSystem extends FileSystem implements IOStatisticsSo
 
   private ITraceFactory traceFactory = TraceFactory.get(/* isEnabled */ false);
 
+  /** Instrumentation to track Statistics */
+  ITraceFactory getTraceFactory() {
+    return this.traceFactory;
+  }
+
   /**
    * Constructs an instance of GoogleHadoopFileSystem; the internal GoogleCloudStorageFileSystem
    * will be set up with config settings when initialize() is called.
    */
-  public GoogleHadoopFileSystem() {}
+  public GoogleHadoopFileSystem() {
+    StorageStatistics globalStats =
+        GlobalStorageStatistics.INSTANCE.put(
+            GhfsGlobalStorageStatistics.NAME, () -> new GhfsGlobalStorageStatistics());
+
+    if (GhfsGlobalStorageStatistics.class.isAssignableFrom(globalStats.getClass())) {
+      globalStorageStatistics = (GhfsGlobalStorageStatistics) globalStats;
+    } else {
+      logger.atWarning().log(
+          "Encountered an error while registering to GlobalStorageStatistics. Some of the GCS connector metrics will not be reported to metrics sinks. globalStatsClassLoader=<%s>; classLoader=<%s>",
+          globalStats.getClass().getClassLoader(),
+          GhfsGlobalStorageStatistics.class.getClassLoader());
+      globalStorageStatistics = GhfsGlobalStorageStatistics.DUMMY_INSTANCE;
+    }
+
+    GoogleCloudStorageEventBus.register(globalStorageStatistics);
+  }
 
   /**
    * Constructs an instance of GoogleHadoopFileSystem using the provided
    * GoogleCloudStorageFileSystem; initialize() will not re-initialize it.
    */
   @VisibleForTesting
   GoogleHadoopFileSystem(GoogleCloudStorageFileSystem gcsfs) {
+    this();
```

Comment:
is this required?

Reply:
Yes this is required to setup global storage statistics